pkg/envtest/server.go

```diff
@@ -221,6 +227,10 @@ func (te *Environment) Start() (*rest.Config, error) {
 			if err != nil {
 				return nil, fmt.Errorf("unable to get configuration for existing cluster: %w", err)
 			}
+			te.KubeConfig, err = controlplane.KubeConfigFromREST(te.Config)
+			if err != nil {
+				return nil, fmt.Errorf("unable to convert configuration to kubeconfig: %w", err)
+			}
```

Comment:
```suggestion

			if len(te.KubeConfig) == 0 {
				te.KubeConfig, err = controlplane.KubeConfigFromREST(te.Config)
				if err != nil {
					return nil, fmt.Errorf("unable to convert configuration to kubeconfig: %w", err)
				}
			}
```
The godoc says this is only populated if not set (which I think makes sense). Let's also implement it that way :)

Reply:
Let's move this down below the if (~ l.305). So that we also get the KubeConfig in the case where we don't use an existing cluster